analyses/cell-type-neuroblastoma-04/exploratory-notebooks/separate-tumor.Rmd

```diff
@@ -0,0 +1,543 @@
+We'll pull out the marker genes for each and ask how similar they are.
+In other words, we'll ask if there is overlap between:
+
+* Marker genes upregulated in `NE` compared to each other cell type
+* Marker genes upregulated in `NE-tumor` compared to each other cell type
+
+```{r}
+# markers upregulated NE compared to other cell types in the separated model
+ne_markers <- model_separated$markers$full$Neuroendocrine
+
+# markers upregulated NE-tumor compared to other cell types in the separated model
+ne.tumor_markers <- model_separated$markers$full$`Neuroendocrine-tumor`
+```
+
+
+Below we print the intersection of these two sets of marker genes for each cell type comparison:
+```{r}
+separated_cell_types <- names(ne_markers)
+non_neuro_cell_types <- separated_cell_types[!(stringr::str_detect(separated_cell_types, "Neuro"))]
+
+intersecting_marker_genes <- non_neuro_cell_types |>
+  purrr::set_names() |>
+  purrr::map(
+    \(cell_type) {
+      intersect(
+        ne_markers[[cell_type]],
+        ne.tumor_markers[[cell_type]]
+      )
+    }
+  )
+
+# Print as data frame so it doesn't print double
+intersecting_marker_genes |>
+  purrr::map(
+    \(x) data.frame(genes = paste(sort(x), collapse = ", "))
+  ) |>
+  purrr::list_rbind(names_to = "cell_type") |>
+  # add column with gene count
+  dplyr::mutate(n_genes = stringr::str_count(genes, ",") + 1) |>
+  dplyr::select(cell_type, n_genes, gene_symbols = genes)
+```
+
+Most marker genes are actually the same for the NE and NE-tumor pairwise comparisons, indeed showing us that these categories are highly similar.
+The main exception is `RBCs` for which there are apparently very different gene lists for NE and NE-tumor. 
+Moreover, the genes tend to be the same across cell types, which suggests these are probably more universal NE markers.
+
+
+
+
+### Expression of different NE marker genes in the separated model
+
+Next, we'll pull out the _distinct_ marker genes associated with `NE` and `NE-tumor` groups that are also not represented in the combined model's `NE` marker genes.
+
+We'll look at these genes' expression in cells that were annotated as `NE` and `NE-tumor` by the separated model.
+We expect that these distributions will show some difference (since `SingleR` will find genes no matter what!) between annotated cell types `NE` and `NE-tumor`.
+If the magnitude of the difference is large, it suggests that separating NE into two groups may not be unreasonable.
+If the magnitude of the difference is small, it suggests that separating NE into two groups is oversplitting.
+
+In the plot, each point represents gene expression in a given cell, and the black diamonds are the mean expression for that gene.
+
+That said, note that there are likely very different numbers of cells labeled as `NE` and `NE-tumor`, which may influence how these plots are interpreted.
+```{r}
+celltype_df |>
+  dplyr::filter(stringr::str_detect(singler_label_separated, "Neuro")) |>
```

Comment:
```suggestion
  dplyr::filter(stringr::str_detect(singler_label_separated, "Neuroendocrine")) |>
```
same here with being specific? I think generally I would just use the full name even though it's a lot of typing.